src/librustc_resolve/diagnostics.rs

```diff
@@ -838,6 +838,31 @@ trait Foo {
 
 fn foo<T>(x: T) {} // ok!
 ```
+
+Another case that causes this error is when a type is imported into a parent
+module. To fix this, you can follow the suggestion and use File directly or
+`use super::File;` which will import the types from the parent namespace. An
+example that causes this error is below:
+
+```
+use std::fs::File;
+
+mod foo {
+    fn some_function(f: File) {}
+}
+```
+
+```
+use std::fs::File;
+
+mod foo {
+    // either
+    use super::File;
```

Comment:
Currently CI failed with 

```
[00:55:49] ---- /checkout/obj/build/x86_64-unknown-linux-gnu/test/error-index.md - Rust_Compiler_Error_Index (line 6486) stdout ----
[00:55:49] 	error[E0432]: unresolved import `super::File`
[00:55:49]  --> <anon>:6:9
[00:55:49]   |
[00:55:49] 6 |     use super::File;
[00:55:49]   |         ^^^^^^^^^^^ no `File` in the root
```

Dunno why, probably workaround it by adding a hidden `# pub` above the `use std::fs::File`?